Microsoft.Toolkit.Uwp.UI.Controls.Core/ImageEx/ImageExBase.Source.cs

```diff
@@ -66,7 +68,11 @@ private static bool IsHttpUri(Uri uri)
             return uri.IsAbsoluteUri && (uri.Scheme == "http" || uri.Scheme == "https");
         }
 
-        private void AttachSource(ImageSource source)
+        /// <summary>
+        /// Method to call to assign an <see cref="ImageSource"/> value to the underlying <see cref="Image"/> powering <see cref="ImageExBase"/>.
+        /// </summary>
+        /// <param name="source"><see cref="ImageSource"/> to assign to the image.</param>
+        protected void AttachSource(ImageSource source)
```

Comment:
This method is called by one of the earlier conditional branches in the method `SetSource` below, but it's not resetting the local `CurrentSourceUri` property. Doesn't this mean that technically the image might load an image synchronously from there while at the same time a previous cached image loading is still asynchronously processing, which would end up overriding the final image once it completes? Since the `CurrentSourceUri` property would still be the same, so that overridden method wouldn't be able to detect a change in the control state 🤔

Reply:
This is handled with the Token.